### PR TITLE
Add support for nullable min/max

### PR DESCRIPTION
**Pull request by faizandfurious**
Adds functions for min/max with pointers, allowing for nullable values.

### CODE REVIEW

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free
```go
if *a > *b {
        return a
}

return b
```

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

File: mathutil.go
Comment (cznic) [inline mathutil.go]: Can be made allocation free

**Review by cznic — CHANGES_REQUESTED** [review]
Thanks for the PR. Please address the inline comments, thank you.

**Comment by faizandfurious** [comment]
Changes have been pushed.

**Comment by bojdell** [comment]
Exciting - thanks for contributing @faizandfurious! This looks good to me, I'll let @cznic have the final approval though

**Review by cznic — APPROVED** [review]
LGTM. Please commit to the master branch, thank you.

**Comment by cznic** [comment]
I believe it was grantad to you before the LGTM. Let me know if not, I'll send the collaborator's invite again (tomorrow). Maybe it fell into the spam folder?

**Comment by faizandfurious** [comment]
👍

**Pull request merged**
